src/components/families/family-sidebar/FamilySidebar.tsx

```diff
@@ -116,16 +129,18 @@ const FamilySidebar = ({ family, isOpen, onClose }: Props) => {
     setIsEditing(editing);
   };
 
-  const handleClose = () => {
-    onToggleEdit(false);
-    onClose();
-  };
-
   const onSubmitFamilyForm = () => {
     // TODO: make PUT request
     setIsEditing(false);
   };
 
+  const debouncedSaveFamily = useCallback(
+    debounce(async (data: FamilyRequest) => {
+      onEditFamily(data);
+    }, 500),
```

Comment:
was the debounce added in case multiple users try to save edits at the same time?

Reply:
it lets us have the effect of saving while editing (like google docs), but throttles the requests so we don't actually make one every time a user enters a character! since i put 500 as the delay, it calls `onEditFamily` 500ms after the last time the notes were edited

can check out more details [in the lodash docs](https://lodash.com/docs/4.17.15#debounce)!